dask_planner/src/expression.rs

```diff
@@ -147,53 +175,24 @@ impl PyExpr {
 
     /// Determines the type of this Expr based on its variant
     #[pyo3(name = "getRexType")]
-    pub fn rex_type(&self) -> RexType {
-        match &self.expr {
-            Expr::Alias(expr, name) => RexType::Reference,
-            Expr::Column(..) => RexType::Reference,
-            Expr::ScalarVariable(..) => RexType::Literal,
-            Expr::Literal(..) => RexType::Literal,
-            Expr::BinaryExpr { .. } => RexType::Call,
-            Expr::Not(..) => RexType::Call,
-            Expr::IsNotNull(..) => RexType::Call,
-            Expr::Negative(..) => RexType::Call,
-            Expr::GetIndexedField { .. } => RexType::Reference,
-            Expr::IsNull(..) => RexType::Call,
-            Expr::Between { .. } => RexType::Call,
-            Expr::Case { .. } => RexType::Call,
-            Expr::Cast { .. } => RexType::Call,
-            Expr::TryCast { .. } => RexType::Call,
-            Expr::Sort { .. } => RexType::Call,
-            Expr::ScalarFunction { .. } => RexType::Call,
-            Expr::AggregateFunction { .. } => RexType::Call,
-            Expr::WindowFunction { .. } => RexType::Call,
-            Expr::AggregateUDF { .. } => RexType::Call,
-            Expr::InList { .. } => RexType::Call,
-            Expr::Wildcard => RexType::Call,
-            _ => RexType::Other,
-        }
+    pub fn rex_type(&self) -> PyResult<RexType> {
+        Ok(self._rex_type(&self.expr))
     }
 
     /// Python friendly shim code to get the name of a column referenced by an expression
     pub fn column_name(&self, mut plan: logical::PyLogicalPlan) -> PyResult<String> {
-        self._column_name(plan.current_node())
+        self._column_name(&plan.current_node())
             .map_err(|e| py_runtime_err(e))
     }
 
     /// Gets the operands for a BinaryExpr call
     #[pyo3(name = "getOperands")]
     pub fn get_operands(&self) -> PyResult<Vec<PyExpr>> {
         match &self.expr {
-            Expr::BinaryExpr { left, op: _, right } => {
-                let mut operands: Vec<PyExpr> = Vec::new();
-                let left_desc: Expr = *left.clone();
-                let py_left: PyExpr = PyExpr::from(left_desc, self.input_plan.clone());
-                operands.push(py_left);
-                let right_desc: Expr = *right.clone();
-                let py_right: PyExpr = PyExpr::from(right_desc, self.input_plan.clone());
-                operands.push(py_right);
-                Ok(operands)
-            }
+            Expr::BinaryExpr { left, op: _, right } => Ok(vec![
```

Comment:
minor nit: we can just ignore attributes we are not interested in

```suggestion
            Expr::BinaryExpr { left, right, .. } => Ok(vec![
```